test/net_http_test.rb

```diff
@@ -0,0 +1,389 @@
+  def test_bulkheads_tickets_are_working
+    with_semian_options(DEFAULT_SEMIAN_OPTIONS) do
+      with_server do
+        ticket_count = Net::HTTP.new("localhost", TOXIC_PORT).raw_semian_options[:tickets]
+        m = Monitor.new
+        acquired_count = 0
+        ticket_count.times do
+          threads << Thread.new do
+            http = Net::HTTP.new("localhost", "#{TOXIC_PORT}")
+            http.acquire_semian_resource(adapter: :nethttp, scope: :connection) do
+              m.synchronize do
+                acquired_count += 1
+              end
+              sleep
+            end
+          end
+        end
+
+        Thread.new do
+          sleep(1) until acquired_count == ticket_count
+          http = Net::HTTP.new("localhost", "#{TOXIC_PORT}")
+          assert_raises Net::ResourceBusyError do
+            http.get("/")
+          end
+          threads.each(&:join)
+        end
+      end
+    end
+  end
+
+  def test_get_is_protected
+    with_semian_options(DEFAULT_SEMIAN_OPTIONS) do
+      with_server do
+        open_circuit!
+        assert_raises Net::CircuitOpenError do
+          Net::HTTP.get(URI("http://localhost:#{TOXIC_PORT}/200"))
+        end
+      end
+    end
+  end
+
+  def test_instance_get_is_protected
+    with_semian_options(DEFAULT_SEMIAN_OPTIONS) do
+      with_server do
+        open_circuit!
+        assert_raises Net::CircuitOpenError do
+          http = Net::HTTP.new("localhost", "#{TOXIC_PORT}")
+          http.get("/")
+        end
+      end
+    end
+  end
+
+  def test_get_response_is_protected
+    with_semian_options(DEFAULT_SEMIAN_OPTIONS) do
+      with_server do
+        open_circuit!
+        assert_raises Net::CircuitOpenError do
+          uri = URI("http://localhost:#{TOXIC_PORT}/200")
+          Net::HTTP.get_response(uri)
+        end
+      end
+    end
+  end
+
+  def test_post_type_1_is_protected
+    with_semian_options(DEFAULT_SEMIAN_OPTIONS) do
+      with_server do
+        open_circuit!
+        assert_raises Net::CircuitOpenError do
+          uri = URI("http://localhost:#{TOXIC_PORT}/200")
+          Net::HTTP.post_form(uri, 'q' => 'ruby', 'max' => '50')
+        end
+      end
+    end
+  end
+
+  def test_http_start_and_inner_methods_are_protected
+    with_semian_options(DEFAULT_SEMIAN_OPTIONS) do
+      with_server do
+        open_circuit!
+
+        uri = URI("http://localhost:#{TOXIC_PORT}/200")
+        assert_raises Net::CircuitOpenError do
+          Net::HTTP.start(uri.host, uri.port) {}
+        end
+
+        close_circuit!
+        Net::HTTP.start(uri.host, uri.port) do |http|
+          open_circuit!
+          get_subclasses(Net::HTTPRequest).each do |action|
+            assert_raises(Net::CircuitOpenError, "#{action.name} did not raise a Net::CircuitOpenError") do
+              request = action.new uri
+              http.request(request)
+            end
+          end
+        end
+      end
+    end
+  end
+
+  def test_custom_raw_semian_options_work
```

Comment:
This should be two tests. One that test the fallback to the default, one that tests